Library/OAuth.swift

```diff
@@ -55,4 +65,67 @@ public struct OAuth {
       return nil
     }
   }
+
+  internal static func handleRedirect(
+    redirectURL url: URL?,
+    error: Error?,
+    verifier: String,
+    onComplete: @escaping (OAuthAuthorizationResult) -> Void
+  ) {
+    guard error == nil else {
+      if let authenticationError = error as? ASWebAuthenticationSessionError,
+        authenticationError.code == .canceledLogin {
+        DispatchQueue.main.async {
```

Comment:
These dispatches are here to prevent us [from releasing Zalgo](https://blog.izs.me/2013/08/designing-apis-for-asynchrony/) - the completion block is _only ever called asynchronously_.